filebeat/fileset/pipelines.go

```diff
@@ -65,16 +66,17 @@ func (reg *ModuleRegistry) LoadPipelines(esClient PipelineLoader, overwrite bool
 			// check that all the required Ingest Node plugins are available
 			requiredProcessors := fileset.GetRequiredProcessors()
 			reg.log.Debugf("Required processors: %s", requiredProcessors)
-			if len(requiredProcessors) > 0 {
+			// APIs do not exist on serverless
+			if len(requiredProcessors) > 0 && !esClient.IsServerless() {
```

Comment:
Just to double check, it is only the API used by `checkAvailableProcessors()` that is unimplemented?

That is this one API call? 
https://github.com/elastic/beats/blob/be28eb445e150ebcd213ece1e206841bb497eb0a/filebeat/fileset/modules.go#L395

That is using https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-info.html#cluster-nodes-info-api-example-ingest

This is only checking that the set of ingest processors we want to use is supported by the node we connected to. If that is the case this seems fine as long as loading the actual ingest pipelines actually works.

Did we test loading ingest pipelines? Attempting to install a module that defines an ingest pipeline is an easy way to test this.

Reply:
Yah, it's just that one API call. And yes, there's tests for pipelines as part of https://github.com/elastic/elastic-agent/pull/3658

Reply:
The existing tests are actually how I figured out this was broken in the first place.